### PR TITLE
Styled personify error message + new membership check logic

### PR DESCRIPTION
**Pull request by retif**

### CODE REVIEW

File: modules/openy_gc_auth/modules/openy_gc_auth_personify/src/Controller/PersonifyAuthController.php [hunk 1]
```diff
@@ -212,25 +215,51 @@ public function signOutUrl(Request $request) {
...
 
-    $memberships = $this->personifyClient->doAPIcall(
-      'GET',
-      'OrderMembershipInformationViews?$filter=ShipMasterCustomerId%20eq%20%27' . $personifyID . '%27&$format=json'
-    );
-    if (!isset($memberships['d'])) {
-      return FALSE;
-    }
+    $body = '<StoredProcedureRequest>
+    <StoredProcedureName>YSV_OpenY_Member_Access</StoredProcedureName>
```
Comment (podarok) [inline modules/openy_gc_auth/modules/openy_gc_auth_personify/src/Controller/PersonifyAuthController.php:225]: So now only Silicon Valley can do SSO?

File: modules/openy_gc_auth/modules/openy_gc_auth_personify/src/Controller/PersonifyAuthController.php [hunk 2]
```diff
@@ -212,25 +215,51 @@ public function signOutUrl(Request $request) {
...
+    $data = $this->personifyClient->doAPIcall('POST', 'GetStoredProcedureDataJSON?$format=json', $body, 'xml');
 
     $isActive = FALSE;
-    foreach ($memberships['d'] as $m) {
-      // "LineStatusDescr"=>"Active", "FulfillStatusDescr"=>"Active|Expired".
-      if ($m['LineStatusDescr'] == 'Active' && $m['FulfillStatusDescr'] == 'Active') {
+
+    if ($data) {
+      $results = json_decode($data['Data'], TRUE);
+      if ($results['Table'][0]['Access'] === 'Approved') {
```
Comment (hamrant) [inline modules/openy_gc_auth/modules/openy_gc_auth_personify/src/Controller/PersonifyAuthController.php:261]: Please add additional condition
```
isset($results['Table'][0]['Access']) && $results['Table'][0]['Access'] === 'Approved'
```

File: modules/openy_gc_auth/modules/openy_gc_auth_personify/src/Controller/PersonifyAuthController.php (hunk 1 above)
Comment (anpolimus) [reply modules/openy_gc_auth/modules/openy_gc_auth_personify/src/Controller/PersonifyAuthController.php:225]: This was discussed between You, me and SV team that we should have the same approach in the checking of membership status by using a custom procedure everywhere.
We Can change it to the OpenY_Member_Access, but in general it was approved as new approach for checking.
If this not a strategy for now, we have to understand how to organize this check.

Comment (podarok) [reply modules/openy_gc_auth/modules/openy_gc_auth_personify/src/Controller/PersonifyAuthController.php:225]: @anpolimus could you share SQL procedure in README then?
I'm fine with it, just wanted to make it flexible

File: modules/openy_gc_auth/modules/openy_gc_auth_personify/src/Controller/PersonifyAuthController.php (hunk 2 above)
Comment (anpolimus) [inline modules/openy_gc_auth/modules/openy_gc_auth_personify/src/Controller/PersonifyAuthController.php:261]:
```suggestion
      if (isset($results['Table'][0]['Access']) ($results['Table'][0]['Access'] === 'Approved')) {
```

File: modules/openy_gc_auth/modules/openy_gc_auth_personify/src/Controller/PersonifyAuthController.php (hunk 1 above)
Comment (anpolimus) [reply modules/openy_gc_auth/modules/openy_gc_auth_personify/src/Controller/PersonifyAuthController.php:225]: I don't have that SQL. I was introduced to API request details only.
I'll ask YMCASV team to share it.

Comment (anpolimus) [reply modules/openy_gc_auth/modules/openy_gc_auth_personify/src/Controller/PersonifyAuthController.php:225]: I've asked them in the letter.

**Pull request merged**